drivers/SmartThings/matter-switch/src/aqara-cube/init.lua

```diff
@@ -0,0 +1,217 @@
+      return true
+  end
+  return false
+end
+
+local callback_timer = function(device)
+  return function()
+    device:emit_event(cubeAction.cubeAction("noAction"))
+  end
+end
+
+local function reset_thread(device)
+  local timer = device:get_field(CUBEACTION_TIMER)
+  if timer then
+    device.thread:cancel_timer(timer)
+    device:set_field(CUBEACTION_TIMER, nil)
+  end
+  device:set_field(CUBEACTION_TIMER, device.thread:call_with_delay(CUBEACTION_TIME, callback_timer(device)))
+end
+
+local function get_field_for_endpoint(device, field, endpoint)
+  return device:get_field(string.format("%s_%d", field, endpoint))
+end
+
+local function set_field_for_endpoint(device, field, endpoint, value, persist)
+  device:set_field(string.format("%s_%d", field, endpoint), value, {persist = persist})
+end
+
+-- The endpoints of each face may increase sequentially, but may increase as in [250, 251, 2, 3, 4, 5]
+-- and the current device:get_endpoints function is valid only for the former so, adds this function.
+local function get_reordered_endpoints(driver, device)
+  if device.network_type ~= device_lib.NETWORK_TYPE_CHILD then
+    local MS = device:get_endpoints(clusters.Switch.ID, {feature_bitmap=clusters.Switch.types.SwitchFeature.MOMENTARY_SWITCH})
+    -- find the default/main endpoint, the device with the lowest EP that supports MS
+    table.sort(MS)
+    if MS[6] < (MS[1] + 150) then
+      -- When the endpoints of each face increase sequentially
+      -- The lowest EP is the main endpoint
+      -- as a workaround, it is assumed that the first endpoint number and the last endpoint number are not larger than 150.
+      return MS
+    else
+      -- When the endpoints of each face do not increase sequentially... [250, 251, 2, 3, 4, 5] 250 is the main endpoint.
+      -- For the situation where a node following these mechanisms has exhausted all available 65535 endpoint addresses for exposed entities,
+      -- it MAY wrap around to the lowest unused endpoint address (refter to Matter Core Spec 9.2.4. Dynamic Endpoint Allocation)
+      local ept1 = {}   -- First consecutive end points
+      local ept2 = {}   -- Second consecutive end points
+      local idx1 = 1
+      local idx2 = 1
+      local flag = 0
+      local previous = 0
+      for _, ep in ipairs(MS) do
+        if idx1 == 1 then
+          ept1[idx1] = ep
+        else
+          if flag == 0
+            and ep <= (previous + 15) then
+            -- the endpoint number does not always increase by 1
+            -- as a workaround, assume that the next endpoint number is not greater than 15
+            ept1[idx1] = ep
+          else
+            ept2[idx2] = ep
+            idx2 = idx2 + 1
+            if flag ~= 1 then
+              flag = 1
+            end
+          end
+        end
+        idx1 = idx1 + 1
+        previous = ep
+      end
+
+      local start = #ept2 + 1
+      idx1 = 1
+      idx2 = start
+      for i=start, 6 do
```

Comment:
I think I understand most of this function but I am a little confused about this section. In an example where the given endpoints are [250, 251, 2, 3, 4, 5], then I believe before this loop, `ept1` would contain [250, 251] and `ept2` would contain [2, 3, 4, 5]. Then, this loop would append the contents of `ept1` to `ept2` so that the function will return the final list of [2, 3, 4, 5, 250, 251]. Is my understanding correct?

Reply:
Sorry for causing confusion. [250, 251, 2, 3, 4, 5] is the order of endpoint numbers that are actually mapped to the Aqara Cube from Face1 to Face6. And, since the logic works after the `table.sort (MS)`, the `ept1` includes [2, 3, 4, 5] and the `ep2` includes [250, 251]. In fact, regardless of whether the `table.sort (MS)` is called or not, this function was required because the result of `device:get_endpoints` is already [2, 3, 4, 5, 250, 251].
In my opinion, if 'device:get_endpoints' is improved for this case, the `local function get_reordered_endpoints (driver, device)` function is not necessary.

Reply:
I see now, makes sense!

Reply:
Ok thank you! I understand now, I think I had it backwards before.